pkg/operator/encryption/state_controller_encryption_test.go

```diff
@@ -50,6 +55,49 @@ func TestEncryptionStateController(t *testing.T) {
 				}
 			},
 		},
+
+		// scenario 2: validates if "encryption-config-kube-apiserver-test" secret with EncryptionConfiguration in "openshift-config-managed" namespace is created,
+		// it also checks the content and the order of encryption providers, this test expects identity first and aescbc second
+		{
+			name:                     "secret with EncryptionConfig is created and it contains a read only key",
+			targetNamespace:          "kms",
+			encryptionSecretSelector: metav1.ListOptions{LabelSelector: "encryption.operator.openshift.io/component=kms"},
+			destName:                 "encryption-config-kube-apiserver-test",
+			targetGRs: map[schema.GroupResource]bool{
+				schema.GroupResource{Group: "", Resource: "secrets"}: true,
+			},
+			initialResources: []runtime.Object{
+				createDummyKubeAPIPod("kube-apiserver-1", "kms"),
+				createSecretBuilder("kms", schema.GroupResource{"", "secrets"}, 1).
+					WithEncryptionKey([]byte("61def964fb967f5d7c44a2af8dab6865")).ToCoreV1Secret(),
+			},
+			// TODO: figure why base64 of 61def964fb967f5d7c44a2af8dab6865 doesn't match
+			//       the test expects NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjUK whereas the actual value is NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjU=
+			//expectedEncryptionCfg: createEncryptionCfgNoWriteKey("1", "NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjUK", "secrets"),
+			expectedEncryptionCfg: createEncryptionCfgNoWriteKey("1", "NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjU=", "secrets"),
+
+			validateFunc: func(ts *testing.T, actions []clientgotesting.Action, destName string, expectedEncryptionCfg *apiserverconfigv1.EncryptionConfiguration) {
+				if len(actions) != 5 {
```

Comment:
We should probably validate actions a bit more specifically.

Reply:
we could validate the verbs, if that's not enough we could validate the verbs and the accompanying resources.

Reply:
I am thinking verb + resource string + resource name/ns always validated and optional full object validation.  Ideally we do this generically in every test case.